File: java/parser/src/main/java/gov/lbl/parser/parser/Component_declarationVisitor.java
package gov.lbl.parser.parser;

import gov.lbl.antlr4.visitor.modelicaBaseVisitor;
import gov.lbl.antlr4.visitor.modelicaParser;
import gov.lbl.parser.domain.Comment;
import gov.lbl.parser.domain.Component_declaration;
import gov.lbl.parser.domain.Condition_attribute;
import gov.lbl.parser.domain.Declaration;

public class Component_declarationVisitor extends modelicaBaseVisitor<Component_declaration> {
    @Override
    public Component_declaration visitComponent_declaration(modelicaParser.Component_declarationContext ctx) {
        DeclarationVisitor declarationVisitor = new DeclarationVisitor();
        Declaration declaration = ctx.declaration().accept(declarationVisitor);
        Condition_attributeVisitor condition_attributeVisitor = new Condition_attributeVisitor();
        Condition_attribute condition_attribute = ctx.condition_attribute() == null ? null : ctx.condition_attribute().accept(condition_attributeVisitor);
        CommentVisitor commentVisitor = new CommentVisitor();
        Comment comment = ctx.comment() == null ? null : ctx.comment().accept(commentVisitor);
        return new Component_declaration(declaration, condition_attribute, comment);
    }
}

File: jsParser/parser/Component_declarationVisitor.js
const modelicaVisitor = require('../antlrFiles/modelicaVisitor').modelicaVisitor
const Component_declaration = require('../domain/Component_declaration')

const DeclarationVisitor = require('./DeclarationVisitor')
const Condition_attributeVisitor = require('./Condition_attributeVisitor')
const CommentVisitor = require('./CommentVisitor')

class Component_declarationVisitor {
  constructor () {
    modelicaVisitor.call(this)
    return this
  }

  visitComponent_declaration (ctx) {
    let declaration = null
    let condition_attribute = null
    let comment = null

    if (ctx.declaration()) {
      const declarationVisitor = new DeclarationVisitor.DeclarationVisitor()
      declaration = declarationVisitor.visitDeclaration(ctx.declaration())
    }
    if (ctx.condition_attribute()) {
      const condition_attributeVisitor = new Condition_attributeVisitor.Condition_attributeVisitor()
      condition_attribute = condition_attributeVisitor.visitCondition_attribute(ctx.condition_attribute())
    }
    if (ctx.comment()) {
      const commentVisitor = new CommentVisitor.CommentVisitor()
      comment = commentVisitor.visitComment(ctx.comment())
    }

    return new Component_declaration.Component_declaration(declaration, condition_attribute, comment)
  }
};

Component_declarationVisitor.prototype = Object.create(modelicaVisitor.prototype)

exports.visitComponent_declaration = this.visitComponent_declaration
exports.Component_declarationVisitor = Component_declarationVisitor
